### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by Alfred-KInyua**
### Data Model Project 
Functions implemented in the "Create Data Model" feature

- [ ] Follow ERD to create tables
- [ ] Foreign Keys and Indexes for them are included
- [ ] Use Migrate syntax to modify the schema.rb file

The following project has implemented the above requirements.

### CODE REVIEW

File: db/migrate/20221019143406_add_user_ref_to_post.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToPost < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
+  end
```
Comment (gemmen29) [inline db/migrate/20221019143406_add_user_ref_to_post.rb]: - [ ] Kindly change the name of the foreign key column name to `author` instead of `user`.

File: db/migrate/20221019143205_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
```
Comment (gemmen29) [inline db/migrate/20221019143205_add_user_ref_to_comments.rb]: - [ ] Kindly change the name of the foreign key column name to `author` instead of `user`.

File: db/migrate/20221019154329_add_user_ref_to_like.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUserRefToLike < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
+  end
```
Comment (gemmen29) [inline db/migrate/20221019154329_add_user_ref_to_like.rb]: - [ ] Kindly change the name of the foreign key column name to `author` instead of `user`.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Alfred-KInyua, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221019154329_add_user_ref_to_like.rb (hunk 3 above)
Comment (Alfred-KInyua) [reply db/migrate/20221019154329_add_user_ref_to_like.rb]: @gemmen29  Thanks for the review I was getting an error when I tried changing, but let me try again.

Comment (gemmen29) [reply db/migrate/20221019154329_add_user_ref_to_like.rb]: @Alfred-KInyua You need to change the value of `foreign key` to point to the user table.

Comment (Alfred-KInyua) [reply db/migrate/20221019154329_add_user_ref_to_like.rb]: @gemmen29 Thanks for your input. have now solved it.

**Review by mikethreels — APPROVED** [review]
Hi Kinyua, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
